gst/nnstreamer/include/tensor_typedef.h

```diff
@@ -232,4 +232,20 @@ typedef struct
   int rate_d; /**< framerate is in fraction, which is numerator/denominator */
 } GstTensorsConfig;
 
+/**
+ * @brief Data structure to describe a tensor data.
+ * This represents the basic information of a memory block for tensor stream.
+ *
+ * Internally NNStreamer handles a buffer with capability other/tensors-flexible using this information.
+ * - version: The version of tensor meta.
+ * - type: The type of each element in the tensor. This should be a value of enumeration tensor_type.
+ * - dimension: The dimension of tensor. This also denotes the rank of tensor. (e.g., [3:224:224:0] means rank 3.)
+ */
+typedef struct
+{
+  uint32_t version;
+  uint32_t type;
+  uint32_t dimension[16];
```

Comment:
Q. `16` means `NNS_TENSOR_SIZE_LIMIT`? Then, you may use this variable.
BTW, why not `NNS_TENSOR_RANK_LIMIT`?

Reply:
No, max dimension for meta info and flex tensor. In offline discussion, supposed that 16 is enough size to describe a tensor. (e.g. flex-tensor)

Reply:
Let's define a macro so that its meaning is exposed.